src/lucky/http2_headers.cr

```diff
@@ -0,0 +1,85 @@
+# HTTP/2 Header Management for Lucky Framework
+#
+# This module provides utilities to ensure HTTP/2 compliance by managing headers
+# with lowercase names and dashes instead of underscores, as required by RFC 7540.
+#
+# The module works around Crystal's header normalization behavior while maintaining
+# compatibility with existing code that expects Title-Case headers.
+module Lucky::HTTP2Headers
```

Comment:
same with this.. Crystal already normalizes all headers, unless there's something here that's HTTP2 specific that I'm not seeing?